forge-gui/res/cardsfolder/upcoming/rocco_street_chef.txt

```diff
@@ -0,0 +1,16 @@
+Name:Rocco, Street Chef
+ManaCost:R G W
+Types:Legendary Creature Elf Druid
+PT:2/4
+T:Mode$ Phase | Phase$ End of Turn | ValidPlayer$ You | TriggerZones$ Battlefield | Execute$ TrigExile | TriggerDescription$ At the beginning of your end step, each player exiles the top card of their library. Until your next end step, each player may play the card they exiled this way.
+SVar:TrigExile:DB$ Dig | Defined$ Player | DigNum$ 1 | ChangeNum$ All | DestinationZone$ Exile | RememberChanged$ True | SubAbility$ DBEffect
+SVar:DBEffect:DB$ Effect | EffectOwner$ RememberedOwner | StaticAbilities$ STPlay | RememberObjects$ Remembered | Duration$ UntilYourNextEndStep | SubAbility$ DBCleanup | ForgetOnMoved$ Exile
```

Comment:
The problem is the duration will be adjusted for each effect controller

Reply:
Hmm any pointers ?

Reply:
I'd need some time to think on it

Because of multiplayer having the effects split is probably the right approach

Reply:
I think a single one might be fine as you should be able to limit it with `MayPlayPlayer$ CardOwner`
but please test

Reply:
Thanks seems to work !